man.vim: Ignore $MANWIDTH and use vim's soft wrap

doronbehar opened the pull request:
Fixes #9017

doronbehar committed:
man.vim: Ignore $MANWIDTH and use vim's soft wrap

doronbehar committed:
man.vim: Ignore $MANWIDTH, use soft wrap #9023

fix #9017
close #9023

justinmk commented on runtime/ftplugin/man.vim:
```diff
@@ -48,5 +48,17 @@ if get(g:, 'ft_man_folding_enable', 0)
+endif
+
+if ! get(g:, 'ft_man_linebreak_disable', 0)
+  setlocal wrap
```
what 'linebreak' setting is needed, if any?

justinmk committed:
man.vim: Fix very long justified lines #9023

When nroff justifies a line, it fills the line with whitespace to meet
$MANWIDTH.  With $MANWIDTH=9999, that of course results in nonsense (and
behaves poorly with 'cursorline' option).

To work around that, instead of trying to hard-justify the lines, just
replace the mega-whitespace with a fixed size of 10 spaces.

Perhaps N/Vim needs a "soft justify" feature?

justinmk commented:
I noticed a problem and pushed an update. See the last commit.

doronbehar committed:
Merge branch 'manwidth0' of github.com:doronbehar/neovim into manwidth0

* 'manwidth0' of github.com:doronbehar/neovim:
  man.vim: Fix very long justified lines #9023
  man.vim: Ignore $MANWIDTH, use soft wrap #9023
  lint
  replace fallthrough comment with macro
  vim-patch:8.0.1215: newer gcc warns for implicit fallthrough

doronbehar committed:
Remove settings controlling soft wrappings

e-kwsm commented:
dc256e3 made the cursor go to the bottom of the manpage.
A simple workaround is to add `exe 'normal! gg'`.

justinmk commented:
@e-kwsm fixed by a1976c739097
